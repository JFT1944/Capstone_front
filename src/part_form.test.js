
import React from "react";
import { render, fireEvent, getByTestId } from "@testing-library/react";
import PartForm from "./Part_Form";
import { BrowserRouter, MemoryRouter } from "react-router-dom";


// write a smoke test for the home component

it('should appear in a smoketest', () => {
    <BrowserRouter>
    render(<PartForm user={'user'} token={'token'} setUser={'setUser'}/>)
    </BrowserRouter>
})

it('should match snapshot', () => {
    const {asFragment} = render(<MemoryRouter><PartForm user={'user'} token={'token'} setUser={'setUser'}/></MemoryRouter>)
    expect(asFragment()).toMatchSnapshot()
})
