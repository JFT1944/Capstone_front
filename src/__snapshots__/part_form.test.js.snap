// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <form
    class="iForms"
  >
    <span>
      Name: 
      <input
        class="input_box"
        name="name"
        placeholder="Name of ingredient"
        type="text"
        value=""
      />
    </span>
    <span>
      Unit: 
      <input
        class="input_box"
        name="unit"
        placeholder="oz, lbs, litre, etc"
        type="text"
        value=""
      />
    </span>
    <span>
      Available Amount: 
      <input
        class="input_box"
        name="available_amount"
        placeholder="How much do you have?"
        type="number"
        value=""
      />
    </span>
    <span>
      Total Amount: 
      <input
        class="input_box"
        name="full_amount"
        placeholder="How much is on the packaging?"
        type="number"
        value=""
      />
    </span>
    <span
      class="submit_button"
    >
      <input
        class="input_box"
        placeholder=""
        type="submit"
        value="Submit Ingredient"
      />
    </span>
  </form>
</DocumentFragment>
`;
